src/promisifyRequest.ts

```diff
@@ -0,0 +1,69 @@
+import { RequestCallback } from './types';
+
+/**
+ * Wraps a method for returning the promise
+ * @param fn a method for wrapping
+ * @param returnObject should return a object with field promise and field with return value of method (Default: `false`)
+ * @returns wrapped method
+ */
+export function promisifyRequest<
+  Result,
+  MethodReturn,
+  Params extends {} | undefined,
+  ReturnObject extends true | false | undefined = undefined,
+>(
+  fn: (params: RequestParams<Result> & Params) => MethodReturn,
+  returnObject?: ReturnObject
+): PromisedRequestMethod<
+  Result,
+  MethodReturn,
+  Omit<Params, RequestCallback>,
+  ReturnObject
+> {
```

Comment:
Maybe refactor this signature to smaller parts?

Reply:
I want to concretize the resulting promised function so i wrote a big signature. Let's keep it as is.